src/which.zig

```diff
@@ -15,28 +15,33 @@ fn isValid(buf: *bun.PathBuffer, segment: []const u8, bin: []const u8) ?u16 {
 // Like /usr/bin/which but without needing to exec a child process
 // Remember to resolve the symlink if necessary
 pub fn which(buf: *bun.PathBuffer, path: []const u8, cwd: []const u8, bin: []const u8) ?[:0]const u8 {
-    if (bun.Environment.os == .windows) {
-        var convert_buf: bun.WPathBuffer = undefined;
-        const result = whichWin(&convert_buf, path, cwd, bin) orelse return null;
-        const result_converted = bun.strings.convertUTF16toUTF8InBuffer(buf, result) catch unreachable;
-        buf[result_converted.len] = 0;
-        std.debug.assert(result_converted.ptr == buf.ptr);
-        return buf[0..result_converted.len :0];
-    }
-    if (bin.len == 0) return null;
 
     // handle absolute paths
     if (std.fs.path.isAbsolute(bin)) {
         bun.copy(u8, buf, bin);
         buf[bin.len] = 0;
         const binZ: [:0]u8 = buf[0..bin.len :0];
+        if (bun.Environment.isWindows) {
+            (std.fs.cwd().openFile(bin, .{}) catch return null).close();
```

Comment:
this will cause a performance issue and a segfault due to their invalid path format error handling code

instead if we need to use a system call here, this should use `GetFileType`

Reply:
actually, `GetFileAttributesW`